libs/api/domains/national-registry/src/lib/national-registry.module.ts

```diff
@@ -0,0 +1,37 @@
+import { DynamicModule } from '@nestjs/common'
+import { NationalRegistryResolver } from './national-registry.resolver'
+import { NationalRegistryService } from './national-registry.service'
+import { NationalRegistryApi } from './soap/nationalRegistryApi'
+import { SoapClient } from './soap/soapClient'
+
+export interface Config {
+  baseSoapUrl: string
+  host: string
+  user: string
+  password: string
+}
+
+export class NationalRegistryModule {
+  static register(config: Config): DynamicModule {
+    return {
+      module: NationalRegistryModule,
+      providers: [
+        NationalRegistryService,
+        NationalRegistryResolver,
+        {
+          provide: NationalRegistryApi,
+          useFactory: async () =>
+            new NationalRegistryApi(
+              await SoapClient.generateClient(
+                config.baseSoapUrl ?? 'https://localhost:8443',
+                config.host ?? 'soffiaprufa.skra.is',
+              ),
+              config.password ?? '',
+              config.user ?? '',
```

Comment:
The defaults here should be specified in the `Config` implementation. From configuration point of view it is better to have all the configs and their defaults specified in one place.